sphinx/tutorial/cylc/furthertopics/cylc-broadcast.rst

```diff
@@ -0,0 +1,132 @@
+Cylc Broadcast
+==============
+
+This tutorial walks you through using cylc broadcast.
+
+This changes :ref:`task runtime configuration <tutorial-runtime>` in a
+running suite, on-the-fly.
+
+
+Purpose
+-------
+
+``cylc broadcast`` can be used to change any ``[runtime]`` setting during a
+suite's run.
+
+You could invoke it on the command line (e.g. as a manual response to a
+failure) or use it in a task script (e.g. as part of an auto-recovery task
+that fixes other tasks).
+
+``cylc broadcast`` can also be used to communicate between tasks, where
+file-based communication is not appropriate. You should not normally need to
+use ``cylc broadcast`` in a task.
+
+
+Standalone Example
+------------------
+
+Create a new suite in the ``cylc-run`` directory called
+``tutorial-broadcast``::
+
+   mkdir ~/cylc-run/tutorial-broadcast
+   cd ~/cylc-run/tutorial-broadcast
+
+Copy the following configuration into a ``suite.rc`` file:
+
+.. code-block:: cylc
+
+   [scheduling]
+       initial cycle point = 1012
+       [[dependencies]]
+           [[[R1]]]
+               graph = wipe_log => announce
+           [[[PT1H]]]
+               graph = announce[-PT1H] => announce
+
+   [runtime]
+       [[wipe_log]]
+           # Delete any files in the suite's "share" directory.
+           script = rm "${CYLC_SUITE_SHARE_DIR}/*"
+
+       [[announce]]
+           script = echo "${CYLC_TASK_CYCLE_POINT} - ${MESSAGE}" >> "${FILE}"
+           [[[environment]]]
+               WORD = ni
+               MESSAGE = We are the knights who say \"${WORD}\"!
+               FILE = "${CYLC_SUITE_SHARE_DIR}/knights"
+
+We now have a suite with an ``announce`` task which runs every hour, writing a
+message to a log file (``share/knights``) when it does so. For the first cycle
+the log entry will look like this::
+
+   10120101T0000Z - We are the knights who say "ni"!
+
+The ``cylc broadcast`` task enables us to change runtime configuration whilst
+the suite is running. For instance we could change the value of the ``WORD``
+environment variable using the command::
+
+   cylc broadcast tutorial-broadcast -n announce -s "[environment]WORD=if"
+
+* ``tutorial-broadcast`` is the name of the suite.
+* ``-n announce`` tells cylc we want to change the runtime configuration of the
+  ``announce`` task.
+* ``-s "[environment]WORD=if"`` changes the value of the ``WORD`` environment
+  variable to ``it``.
+
+Run the suite then try using the ``cylc broadcast`` command to change the
+message::
+
+   cylc run tutorial-broadcast
+   cylc broadcast tutorial-broadcast -n announce -s "[environment]WORD=it"
+   ...
```

Comment:
Why the ellipsis here? If you are suggesting running the command on the second line twice maybe repeat that line with a different 'WORD=x' example on a new line before the final ellipsis line as it is not clear (at least to me) quite what is intended.

Reply:
I'm not sure either, gone.